nix/cc/nixpkgs_cc_libraries.bzl

```diff
@@ -0,0 +1,109 @@
+""" Bazel cc_library definitions for Nixpkgs. """
+
+load("@io_tweag_rules_nixpkgs//nixpkgs:nixpkgs.bzl", "nixpkgs_flake_package")
+
+def nixpkgs_cc_library_deps():
+    """ Generate cc_library rules for Nixpkgs. """
+    return [
+        nixpkgs_flake_package(
+            name = "org_openssl_%s" % system,
+            nix_flake_file = "//:flake.nix",
+            nix_flake_lock_file = "//:flake.lock",
+            package = "packages.%s.openssl" % system,
+            build_file_content = OPENSSL_BUILD,
+        )
+        for system in openssl_systems
+    ] + [
+        nixpkgs_flake_package(
+            name = "cryptsetup_%s" % system,
+            nix_flake_file = "//:flake.nix",
+            nix_flake_lock_file = "//:flake.lock",
+            package = "packages.%s.cryptsetup" % system,
+            build_file_content = CRYPTSETUP_BUILD,
+        )
+        for system in cryptsetup_systems
+    ] + [
+        nixpkgs_flake_package(
+            name = "libvirt_%s" % system,
+            nix_flake_file = "//:flake.nix",
+            nix_flake_lock_file = "//:flake.lock",
+            package = "packages.%s.libvirt" % system,
+            build_file_content = LIBVIRT_BUILD,
+        )
+        for system in libvirt_systems
+    ]
+
+openssl_systems = [
+    "aarch64-linux",
+    "aarch64-darwin",
+    "x86_64-linux",
+    "x86_64-darwin",
+]
+
+cryptsetup_systems = [
+    "aarch64-linux",
+    "x86_64-linux",
+]
+
+libvirt_systems = [
+    "aarch64-linux",
+    "x86_64-linux",
+]
+
+OPENSSL_BUILD = """\
+load("@rules_cc//cc:defs.bzl", "cc_library")
+filegroup(
+    name = "include",
+    srcs = glob(["include/**/*.h"]),
+    visibility = ["//visibility:public"],
+)
+cc_library(
+    name = "org_openssl",
+    srcs = glob(["lib/**/*.a"]),
+    hdrs = [":include"],
+    strip_include_prefix = "include",
+    visibility = ["//visibility:public"],
+)
+"""
+
+CRYPTSETUP_BUILD = """\
+filegroup(
+    name = "include",
+    srcs = glob(["include/**/*.h"]),
+    visibility = ["//visibility:public"],
+)
```

Comment:
Could we reuse the built-in targets https://github.com/tweag/rules_nixpkgs/blob/21c4ea481021cb51a6e5d0969b2cee03dba5a637/core/BUILD.bazel.tpl#L14?

Reply:
I think we need to customize our `BUILD` file to incorporate the `cc_library` and `export_files` rules.
nixpkgs will also only provide the built-in targets if `build_file_content` is not set:

https://github.com/tweag/rules_nixpkgs/blob/21c4ea481021cb51a6e5d0969b2cee03dba5a637/core/nixpkgs.bzl#L474-L482

Reply:
Understood